Model/Cc.php

```diff
@@ -217,6 +217,10 @@ public function __construct(
         $this->errorHandler = $errorHandler;
     }
 
+    /** Why do we need both isAvailable and isActive
```

Comment:
so this is where the /getPaymentMethods API call would help. We need to know that Paypal, for example, is available for this customer and check that it's enabled for this magento instance.

So we do need both, but the way that it is written right now is not correct for credit cards. We would need to check at the specific card level, not just credit cards in general.  We'll have to talk through this to outline the details before changing the way this code is functioning right now.

Reply:
Yes agree. Thanks!